pandas/core/groupby.py

```diff
@@ -4429,7 +4429,13 @@ def _groupby_indices(values):
         # bit better than factorizing again
         reverse = dict(enumerate(values.categories))
         codes = values.codes.astype('int64')
-        _, counts = _hash.value_count_int64(codes, False)
+        keys, counts = _hash.value_count_int64(codes, False)
```

Comment:
Rather than adjusting the hash table value counts, I think you could use `np.bincount` here? - see  `Categorical.value_counts`